libraries/redcore/api/hal/hal.php

```diff
@@ -1262,7 +1262,12 @@ public function processPostData($data, $configuration)
 				$fieldAttributes = RApiHalHelper::getXMLElementAttributes($field);
 				$fieldAttributes['transform'] = !is_null($fieldAttributes['transform']) ? $fieldAttributes['transform'] : 'string';
 				$fieldAttributes['defaultValue'] = !is_null($fieldAttributes['defaultValue']) ? $fieldAttributes['defaultValue'] : '';
-				$data[$fieldAttributes['name']] = !is_null($data[$fieldAttributes['name']]) ? $data[$fieldAttributes['name']] : $fieldAttributes['defaultValue'];
+
+				if (!isset($data[$fieldAttributes['name']]) || is_null($data[$fieldAttributes['name']]))
```

Comment:
I think it's still need, because if just use the isset function, the logic could run wrong.
The logic is need 2 case:
1. The parameter is not provide => set default value
2. The parameter is provide but not have value => set default value

If we remove the second check, somehow the parameter could exist but don't have any value and it's doesn't set default value as we want.